doc/chapters/api/index.rst

```diff
@@ -3,131 +3,230 @@
 API
 ***
 
-This chapter documents the D-Bus API for interacting with the
-SoftwareContainerAgent. This is mostly useful for integrators.
+This chapter documents the D-Bus API for interacting with the SoftwareContainerAgent. This is mostly
+useful for integrators, or anyone writing a launcher for running apps inside SoftwareContainer
+containers.
 
 .. _dbus-api:
 
 D-Bus API
 =========
 
-D-Bus API is an IPC interface to call SoftwareContainer agent methods. The API
-provides following object path and interface.
+The D-Bus API is an interface to call SoftwareContainer agent methods. The API provides following
+object path and interface.
 
-:Object Path: /com/pelagicore/SoftwareContainer
-:Interface: com.pelagicore.SoftwareContainerAgent
+* Object Path: /com/pelagicore/SoftwareContainerAgent
+* Interface: com.pelagicore.SoftwareContainerAgent
 
 Error Handling
 --------------
-All methods in this API can and will return D-Bus errors on failure. All calls
-that don't return a D-Bus error have worked successfully. Any call that returns
-a D-Bus error has failed. All calls that change a state and raise a D-Bus error
-shall have their side effects documented in this chapter.
+All methods in this API can and will return D-Bus errors on failure. All calls that don't return a
+D-Bus error have worked successfully. Any call that returns a D-Bus error has failed. All calls that
+change a state and raise a D-Bus error shall have their side effects documented in this chapter.
+
+For each method, we describe the various error types and their potential sources. Some of these
+errors have a note saying that they put the container in an `invalid state`. An invalid state means
+that no other operations can be reliably done, and SoftwareContainer will disallow any operations on
+a container that is in an invalid state, except for destroying them.
 
 Methods
 -------
 
 BindMount
 ---------
```

Comment:
Do these two headers end up on the same level? It looks like the same level. Is it possible to get them on different levels?